ext/mbstring/mbstring.c

```diff
@@ -2282,6 +2289,280 @@ PHP_FUNCTION(mb_output_handler)
+	size_t mb_chunk_size; /* chunk size in chars */
+	size_t split_length; /* defined chunk size in chars */
+	mbfl_convert_filter *next_filter; /* widechar to encoding filter */
+};
+
+/* callback function to fill split array */
+static int mbfl_split_output(int c, void *data)
+{
+	struct mbfl_split_params *params = (struct mbfl_split_params *)data; /* cast passed data */
+
+	(*params->next_filter->filter_function)(c, params->next_filter); /* decoder filter */
+
+	size_t mb_chunk_size = ++params->mb_chunk_size; /* increment chunk size */
+	if(params->split_length == mb_chunk_size) { /* if current chunk size reached defined chunk size */
+		mbfl_convert_filter_flush(params->next_filter);/* concatenate separate decoded chars to the solid string */
+		mbfl_memory_device *device = (mbfl_memory_device *)params->next_filter->data; /* chars container */
+		mbfl_string *chunk = params->result_string;
+		mbfl_memory_device_result(device, chunk); /* make chunk */
+		add_next_index_stringl(params->return_value, chunk->val, chunk->len); /* add chunk to the array */
+		efree(chunk->val);
+		params->mb_chunk_size = 0; /* reset mb_chunk size */
+	}
+	return 0;
+}
+
+PHP_FUNCTION(mb_str_split)
+{
+	zend_string *str, *encoding = NULL;
+	size_t mb_len, chunks, chunk_len;
+	char unsigned const *last, *p;
+	mbfl_string string, result_string;
+	zend_long split_length = 1;
+
+	ZEND_PARSE_PARAMETERS_START(1, 3)
+		Z_PARAM_STR(str)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(split_length)
+		Z_PARAM_STR(encoding)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (split_length <= 0) {
+		php_error_docref(NULL, E_WARNING, "The length of each segment must be greater than zero");
+		RETURN_FALSE;
+	}
+
+	/* fill mbfl_string structure */
+	string.no_language = MBSTRG(language);
+	string.encoding = php_mb_get_encoding(encoding);
+	if (!string.encoding) {
+		RETURN_FALSE;
+	}
+
+	p = (char unsigned *)ZSTR_VAL(str); /* string cursor pointer */
+	string.val = (char unsigned *)p;
+	string.len = ZSTR_LEN(str);
+	last = string.val + string.len;
+
+
+	/*
+	 * +----------------------------------------------------------------------+
+	 * |                           4 scenarios                                |
+	 * +----------------------------------------------------------------------+
+	 */
+	const mbfl_encoding *mbfl_encoding = string.encoding;
+
+	/*
+	 * +----------------------------------------------------------------------+
+	 * | first scenario: "fixed width encodings"                              |
+	 * +----------------------------------------------------------------------+
+	 */
+	if (mbfl_encoding->flag & MBFL_ENCTYPE_SBCS) { /* 1 byte */
+		mb_len = string.len;
+		chunk_len = (size_t)split_length; /* chunk length in bytes */
+
+	} else if (mbfl_encoding->flag & (MBFL_ENCTYPE_WCS2BE | MBFL_ENCTYPE_WCS2LE )) { /* 2 bytes */
+		mb_len = string.len>>1; /* string length in chars */
+		chunk_len = split_length<<1; /* chunk length in bytes */
```

Comment:
I think using division and multiplication operators instead of bitshifts would be more readable here.

Reply:
I agree that >> << is less readable than / *, but I still do not want to change. Comments would be enough.